### PR TITLE
[draft] Update Jobs index to watch store, not model

### LINKED ISSUE #10555: ui: Jobs list doesn’t show a new job (on a locally-built binary?)
### Nomad version

```
Nomad v1.1.0-dev (b890a1f7eb298d25fcb39fd2adbc9fe3d8ee14c2+CHANGES)
```

### Operating system and Environment details

macOS 10.14.6

### Issue

When viewing the Nomad UI at `localhost:4646` on a locally-built binary, the jobs list doesn’t show a job newly submitted elsewhere. It’s not happening with a released binary 🤔

### Reproduction steps

Build a Nomad binary that includes the UI with `make dev-ui`, run it, open the jobs list at `localhost:4646/ui/jobs`, submit a job via CLI.

#### Expected Result

The job should appear in the jobs list.

#### Actual Result

The job only appears upon refresh, or by leaving and returning to the jobs list.

jobs-no-live-reload

### PR DESCRIPTION
**Pull request by philrenaud**
Resolves #10555 

Looking for a review in spite of draft state, just to make sure I'm not way off-course.

Problem: Our watcher system's query method doesn't adequately delete or repopulate the store when jobs are started or stopped+purged.

This PR adds some logic to ensure that the query method adequately deletes a record of a purged entity, and disregards the route model once within the controller, opting to query against the live-updating Ember Data store instead.

### CODE REVIEW

File: ui/app/adapters/watchable.js [hunk 1]
```diff
@@ -98,6 +98,8 @@ export default class Watchable extends ApplicationAdapter {
       additionalParams,
       query
     );
+    // If your query is effectively a findAll under the hood, treat non-returned entities as deleted
+    const isPseudoFindAll = Object.values(params).every((v) => v === '*');
```
Comment (ChaiWithJai) [inline ui/app/adapters/watchable.js:102]: `question`: The callback function in the `every` method assumes that `*` wildcard is applicable... However, I'm wondering the `types` of this params object specifically `queryString.parse(params)`.

```
  queryStringParams // based on URL in the browser which will create an object like {token: 'bada55cafe'}
```

Wouldn't we need a recursive operation here? Also note that if any query parameter that is not `namespace` is added then we can exit early knowing that it is not a `psuedoFindAll` operation.

File: ui/app/controllers/jobs/index.js [hunk 2]
```diff
@@ -191,10 +191,15 @@ export default class IndexController extends Controller.extend(
     Visible jobs are those that match the selected namespace and aren't children
     of periodic or parameterized jobs.
   */
-  @computed('model.jobs.@each.parent')
+  @computed('model.jobs.@each.parent', 'qpNamespace', 'store', 'storedJobs.[]')
   get visibleJobs() {
     if (!this.model || !this.model.jobs) return [];
-    return this.model.jobs
+    return this.store
+      .peekAll('job')
```
Comment (ChaiWithJai) [inline ui/app/controllers/jobs/index.js:198]: `question`: Should we use `storedJobs` here so we don't call the store again? I'm also noticing we're declaring `storedJobs.[]` as a dependency here, however `storedJobs` isn't a reactive property. What the benefit of this?

File: ui/app/routes/jobs/index.js [hunk 3]
```diff
@@ -32,7 +32,7 @@ export default class IndexRoute extends Route.extend(
     controller.set('namespacesWatch', this.watchNamespaces.perform());
     controller.set(
       'modelWatch',
-      this.watchJobs.perform({ namespace: controller.qpNamesapce })
+      this.watchJobs.perform({ namespace: controller.qpNamespace })
```
Comment (ChaiWithJai) [inline ui/app/routes/jobs/index.js:35]: `praise`: Great catch here, buddy!

**Review by ChaiWithJai — COMMENTED** [review]
Leaving a first round of review here before I pull the PR down and build it locally. Will do one more round of review before we meet later today.

File: ui/app/adapters/watchable.js [hunk 4]
```diff
@@ -125,17 +127,34 @@ export default class Watchable extends ApplicationAdapter {
...
+            // Special consideration for * queries, like "All jobs with * namespace":
+            // entities will generally have "default" instead of "*" as their value for these sorts of properties.
+            const queryValue =
+              query[mapping.queryParam] === '*'
+                ? 'default'
+                : query[mapping.queryParam];
+            return (
+              get(record, mapping.attr) === queryValue ||
+              get(get(record, mapping.attr), 'id') === queryValue
+            );
```
Comment (ChaiWithJai) [inline ui/app/adapters/watchable.js:143]: This is a great refactoring. I'd like to discuss this further and how it'll impact the work for [15008](https://github.com/hashicorp/nomad/issues/15008) to unblock [14787](https://github.com/hashicorp/nomad/issues/14787).

File: ui/app/adapters/watchable.js [hunk 5]
```diff
@@ -125,17 +127,34 @@ export default class Watchable extends ApplicationAdapter {
...
         .peekAll(type.modelName)
-        .filter((record) =>
-          queryParamsToAttrs.some(
-            (mapping) => get(record, mapping.attr) === query[mapping.queryParam]
-          )
-        )
-        .forEach((record) => {
-          removeRecord(store, record);
+        .filter((record) => {
+          return queryParamsToAttrs.some((mapping) => {
```
Comment (ChaiWithJai) [inline ui/app/adapters/watchable.js:133]: `question`: Why are we filtering against `queryParamsToAttrs`?

**Comment by philrenaud** [comment]
Closing in favour of alternate method of handling

**Pull request closed without merging**